### PR TITLE
find repetitive regions in reference

### PR DESCRIPTION
**Pull request by brentp**
reads that map perfectly to non-repetetive regions
of the genome can't be repetitive so we can skip
the repeat calculation on them.

this improves the reads/sec from ~120K/sec to ~800K/sec
after an initial 4 minute penalty of calculating the reference
repeat content

it can also write the str content to a file (and then re-use it) so that step is
only needed once.

### CODE REVIEW

**Comment by hdashnow** [comment]
```
2.72s$ nimble install -y
  Verifying dependencies for str@0.0.1
    Prompt: No local packages.json found, download it from internet? -> [forced yes]
Downloading Official package list
    Success Package list downloaded.
 Installing kmer@>= 0.2.2
Downloading https://github.com/brentp/nim-kmer using git
       Tip: 4 messages have been suppressed, use --verbose to show them.
     Error: Downloaded package's version does not satisfy requested version range: wanted >= 0.2.2 got 0.2.1.
The command "nimble install -y" failed and exited with 1 during .
```

**Comment by brentp** [comment]
should be ok now. forgot to commit that file in nim-kmer.

**Comment by hdashnow** [comment]
`... /home/travis/build/hdashnow/str-dev/src/strpkg/genome_strs.nim(5, 8) Error: cannot open file: lapper`

**Comment by hdashnow** [comment]
Still failing: https://travis-ci.com/hdashnow/str-dev/builds/126490091

File: src/strpkg/extract.nim [hunk 1]
```diff
@@ -15,42 +16,21 @@ import strformat
...
-      # bail if we do not. this is an optimization to avoid counting when we
-      # cant possibly see a repeat.
-      break
-
-proc get_repeat*(aln:Record, counts:var Seqs[uint8], repeat_count: var int, align_length: var int, opts:Options): array[6, char] =
+proc get_repeat*(aln:Record, genome_str:TableRef[string, Lapper[region]], counts:var Seqs[uint8], repeat_count: var int, align_length: var int, opts:Options): array[6, char] =
   when defined(skipFullMatch):
-    # TODO: write the code to make find repetitive regions in fasta and use
-    # that with this as default.
     # compile with -d:skipFullMatch to make it much faster
```
Comment (hdashnow) [inline src/strpkg/extract.nim]: Is there an advantage to having this option set at compile rather than as a command line option?

Comment (brentp) [reply src/strpkg/extract.nim]: this could be removed as the new genome (fasta) scan works almost as fast as this.

**Pull request merged**